### ISSUE TITLE
Bug: Changing the current word under the cursor broken

### ISSUE THREAD

**Issue by DMunkei** [smjonas/inc-rename.nvim#45]
Hey, when I use the snippet provided in the config to rename a variable it works, as long as I don't completely remove the previous variable.  Hopefully the video explains it better.

Also an aside question, does this use the AST for renaming?

https://github.com/smjonas/inc-rename.nvim/assets/17084284/3d2250b3-f97e-448e-ab75-a9dc403ed8be

This is how my config looks like right now. I'm using Lazy as my plugin manager

```lua
return {
  "smjonas/inc-rename.nvim",
  config = function()
    require("inc_rename").setup()
    vim.keymap.set("n", "<leader>rn", function()
      return ":IncRename " .. vim.fn.expand("<cword>")
    end, { expr = true })
  end,
}
```
This is the neovim version I'm running 
```bash
➜ nvim --version
NVIM v0.9.1
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
```

**Comment by DMunkei** [comment]
I was using Pyright and jedi-language-server.

**Comment by pmartinsdev** [comment]
Hy guys, any date to fix that issue ??  I had the same problem here using the  "js/ts language server" at lazyvim

**Comment by DMunkei** [comment]
I've also had this issue constantly happening inside of TS while writing vuejs code. It somehow just deletes more than it should. Not sure why.

**Comment by smjonas** [comment]
> Hy guys, any date to fix that issue ?? I had the same problem here using the "js/ts language server" at lazyvim

@pmartinsdev Thanks for also reporting this issue. Can you please provide me with example code where this is happening? It's hard for me find out what the problem is without that.

**Comment by Moerliy** [comment]
I mentioned this in #53 too. 
> Something I noticed and could have the same root problem:
>     1. Same project, same file
>     2. line 92 and try renaming ParseContext.
>     3. Add a couple of characters and have a look at the preview. With every new character added to the rename, the previously added characters should appear again with every new character you add. Hard to describe but the number of characters shown in the preview is the factorial of the number of characters added to the rename.
>     (not sure if you can reproduce it tho because that behavior is a bit less predictable)

You should be able to reproduce this with the second example.

**Comment by smjonas** [comment]
Did anyone still observe this issue? If not, this issue can be closed

**Comment by DMunkei** [comment]
I haven't used the pluggin in a while. I can try it out tomorrow and report if the issue persists. However I am not using the same machine as the one I reported the issue in.

**Comment by DMunkei** [comment]
I haven't used the pluggin in a while. I can try it out tomorrow and report if the issue persists. However I am not using the same machine as the one I reported the issue in.

**Comment by unphased** [comment]
I get this issue too!!! Started recently. Might have been from when i built neovim from source to update it recently. 

```
NVIM v0.11.0-dev-782+ga0d8c2b86-Homebrew
Build type: Release
LuaJIT 2.1.1725453128
```

**Comment by Ceres6** [comment]
I'm having the same issue

**Comment by leleuvilela** [comment]
I'm having the same issue here. :')

**Comment by smjonas** [comment]
I finally managed to reproduce the issue by installing `vtsls` and `ts_ls` language servers and performing the rename in a typescript project. The issue was that the `textDocument/rename` would be sent to multiple capable clients which would each rename the variable. It should be fixed in the latest commit. Thank you @leleuvilela for the `:LspInfo` output and thanks you to everyone for your reports and patience!

**Comment by DMunkei** [comment]
Could this also happen when you have multiple LSP running for another language?

I have ruff, basedpyright and jedi for Python.

**Comment by smjonas** [comment]
@DMunkei yes, I think so. It shouldn't be Typescript-specific but could occur every time multiple LSPs are active at the same time in a buffer.

**Comment by DMunkei** [comment]
So is the fix to disable the rename capabilities for all except one?

**Comment by DMunkei** [comment]
I just tried it out and it seems to be working fine in python :)
